fluent-syntax/src/ftlstream.js

```diff
@@ -262,13 +278,20 @@ export class FTLParserStream extends ParserStream {
     }
   }
 
-  takeIDStart() {
-    if (this.isIDStart()) {
+  takeIDStart(allowPrivate) {
+    if (allowPrivate && this.currentIs('-')) {
+      this.next();
+      return '-';
+    }
```

Comment:
You should be able to `return this.next();` here.

Reply:
We want to return `current` here but first, we advance the `index`.